testif.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+# This script is invoked by Travis testing and inspects
+# the changeset that's being built.
+#
+# Usage:
+#
+#    testif.sh <label> && something_else
+#
+# The intention is to be able to ONLY test certain aspects
+# when special conditions are satisfied.
+#
+# Notes:
+#    JS tests are expected to be very fast
+#    We shouldn't skip tests when it affects Coverage negatively
+#    We can possibly do tests based on which branch is pushed to
+#
+# https://github.com/learningequality/kolibri/issues/27
+#
+# What changes are analyzed?
+#
+# For this script, we should mainly use Travis' env
+# $TRAVIS_COMMIT_RANGE. The reason is simple: You cannot tell what
+# changes have been implemented since last test, thus if your
+# condtion should fire or not.
+#
+# Definition:
+# TRAVIS_COMMIT_RANGE: The range of commits that were included in
+# the push or pull request.
+#
+# https://docs.travis-ci.com/user/environment-variables/
+
+
+set -e
+
+LABEL="$1"
+
+commit="$TRAVIS_COMMIT"
+commits="$TRAVIS_COMMIT_RANGE"
+git_changeset=`git show --name-only --no-notes --oneline $commits`
+
+
+# If something changes that's related to our sdist packaging
+# or installation mechanism, then we build and install.
+if [[ "$LABEL" == "setup_changed" ]]
+then
+
+    # Match with commit messages containing "[ setup ]"
+    # Match commits changing requirements.txt
+    # Match commits changing setup.py
+    if echo "$git_changeset" | grep -q "\[\s*setup\s*\]" || \
```

Comment:
1. Should we match commit messages like this? I think it would be better to match just specific files, since "setup" could be used in a more general sense too.
2. I looked at the output of the `git_changeset` command, and my commit messages aren't surrounded by `[` & `]` anyway, so I don't think this would match it...?